### ISSUE TITLE
[CLI] Using a non-full backward hook leads to user warning.

### ISSUE THREAD

**Issue by wzhang2022** [wandb/client#2027]
**Description**
A clear and concise description of the bug

**Wandb features**
`wandb.watch` and `wandb.init`.

**How to reproduce**
Run this:
```
import torch
import torch.nn as nn
import wandb

with wandb.init(project="PerceiverProject"):
    model = nn.Sequential(nn.Linear(100, 50),
                          nn.ReLU(),
                          nn.Linear(50, 5))
    wandb.watch(model)
    model.train()
    img = torch.randn(5, 100) # 1 imagenet image, pixelized
    criterion = nn.CrossEntropyLoss(reduction="mean")
    labels = torch.Tensor([0,1,2,3,4]).long()
    output = model(img)
    loss = criterion(output, labels)
    loss.backward()
```
Warning:
```
C:\Users\Me\anaconda3\envs\Project\lib\site-packages\torch\nn\modules\module.py:795: UserWarning: Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior.
  warnings.warn("Using a non-full backward hook when the forward contains multiple autograd Nodes "
```

**Environment**
- OS: Windows 10
- Python Version: 3.8.8
- PyTorch Version: 1.8.0

**Comment by cvphelps** [comment]
HI @wzhang2022 thank you for reporting this issue. Did you see anything missing in the logged gradients in the UI, or was this just a warning?

**Comment by wzhang2022** [comment]
I only see this warning. I'm not sure what the warning means or how to avoid it. Given that wandb is using a feature of PyTorch that "is deprecated and will be removed in future versions," it seems like this is worth investigating.

**Comment by nabito** [comment]
By using wand.watch with Pytorch I can confirm see this warning as well:

"Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior."

It only appears when I'm training the model.
Not sure if it's going to affect the model training correctness, so I'm opt out from using wandb.watch for now..

**Comment by fishbotics** [comment]
Ditto here. I see this when using wandb_logger.watch in Pytorch lightning.

**Comment by borisdayma** [comment]
This should now be fixed with https://github.com/wandb/client/pull/2077

**Comment by vanpelt** [comment]
The fix actually got pushed to the next release.  If you want to try the fix before we release next week, you can install wandb with:

`pip install --upgrade git+git://github.com/wandb/client.git@bug/issue-1122-2#egg=wandb`
